packages/iov-keycontrol/src/keyring.ts

```diff
@@ -187,4 +188,23 @@ export class Keyring {
   private getMutableWallet(id: WalletId): Wallet | undefined {
     return this.wallets.find(wallet => wallet.id === id);
   }
+
+  /**
+   * Throws if any of the new identities already exists in this keyring.
+   */
+  private ensureNoIdentityCollision(newIdentities: ReadonlyArray<PublicIdentity>): void {
+    const existingIdentities = this.wallets.reduce((currentList, wallet) => {
+      return currentList.concat(wallet.getIdentities());
+    }, new Array<PublicIdentity>());
+
+    for (const newIdentity of newIdentities) {
```

Comment:
Slightly odd combination of a `.reduce` and a `for` loop.

Reply:
Yeah, I only use reduce because our linter say variables must be immutable and to use ReadonlyArray. But I find it very hard to read. Will try to pull out a getAllIdentities function that uses a mutable array

Reply:
I mean, the other option would be to do the comparison in the `reduce` loop.